packages/advisor-components/src/RuleDescription/RuleDescription.tsx

```diff
@@ -0,0 +1,35 @@
+import './RuleDescription.scss';
+
+import React from 'react';
+import Markdown from 'react-markdown';
+import rehypeRaw from 'rehype-raw';
+import rehypeSanitize from 'rehype-sanitize';
+
+import { AdvisorProduct, RuleContentOcp, RuleContentRhel } from '../types';
+
+interface RuleDescriptionProps {
+  product: AdvisorProduct;
+  rule: RuleContentOcp | RuleContentRhel;
+  isDetailsPage: boolean;
+}
+
+const RuleDescription: React.FC<RuleDescriptionProps> = ({ product, rule, isDetailsPage }) => {
+  const ruleDescription = (data: string, isGeneric = false) => (
+    <React.Fragment>
+      {data && (
+        <span className={isGeneric ? 'ins-c-generic__override' : ''}>
+          <Markdown rehypePlugins={[rehypeRaw, rehypeSanitize]}>{data}</Markdown>
+        </span>
+      )}
+    </React.Fragment>
+  );
+
+  if (product === AdvisorProduct.ocp) {
+    return ruleDescription(rule.generic, true);
+  } else {
+    // RHEL rule
+    return isDetailsPage ? ruleDescription(rule.generic, true) : ruleDescription(rule.summary);
+  }
+};
```

Comment:
We can further improve this by returning directly the component, you can even remove the `return` and return this directly.

```suggestion
const RuleDescription: React.FC<RuleDescriptionProps> = ({ product, rule, isDetailsPage }) => (
    <React.Fragment>
      {(rule.generic || rule.summary) && (
        <span className={(product === AdvisorProduct.ocp || isDetailsPage) ? 'ins-c-generic__override' : ''}>
          <Markdown rehypePlugins={[rehypeRaw, rehypeSanitize]}>
            {(product === AdvisorProduct.ocp || isDetailsPage) ? rule.generic : rule.summary }
          </Markdown>
        </span>
      )}
    </React.Fragment>
  );
```